gnovm/pkg/gnolang/preprocess.go

```diff
@@ -968,9 +968,21 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 								}
 							}
```

Comment:
To fix it, all we need to do is to add `convertConst(store, last, arg0, ct)` after lineL969. We don't need the rest of changes. 

we convert untyped const to target type instead of to default type.

Reply:
This was the first thing I tried in commit [dd7f805](https://github.com/gnolang/gno/pull/1423/commits/dd7f805fbb407e546fac146869ff49236320c52d). It breaks a lot of tests due to it actually doing non implicit conversions between types that are not otherwise compatible.

You can see some of the failing tests in the output of this action: https://github.com/gnolang/gno/actions/runs/7135334253

Reply:
we are looking at the different places.

Here is the code 
```

	                                       switch ct.Kind() {
						case IntKind, Int8Kind, Int16Kind, Int32Kind, Int64Kind,
							UintKind, Uint8Kind, Uint16Kind, Uint32Kind, Uint64Kind,
							BigintKind:
							if bd, ok := arg0.TypedValue.V.(BigdecValue); ok {
								if !isInteger(bd.V) {
									panic(fmt.Sprintf(
										"cannot convert %s to integer type",
										arg0))
								}								
				L969		   } 
		                L970              convertConst(store, last, arg0, ct)  <<<< add here. 
					        }
					    // (const) untyped decimal -> float64.
					   // (const) untyped bigint -> int.
					    convertConst(store, last, arg0, nil) 
					   // evaluate the new expression.
					   cx := evalConst(store, last, n)
```

Reply:
You're right; good catch. I've modified the solution.  Please let me know if it looks good now.